aviary/subsystems/propulsion/utils.py

```diff
@@ -104,6 +107,54 @@ def convert_geopotential_altitude(altitude):
     return altitude
 
 
+def build_engine_deck(aviary_options: AviaryValues):
+    '''
+    Creates an EngineDeck using avaliable inputs and options in aviary_options.
+
+    Parameter
+    ----------
+    aviary_options : AviaryValues
+        Options to use in creation of EngineDecks.
+
+    Returns
+    ----------
+    engine_models : <list of EngineDecks>
+        List of EngineDecks created using provided aviary_options.
+    '''
+    # import locally placed to avoid circular import
+    from aviary.subsystems.propulsion.engine_deck import EngineDeck
+
+    # Build a single engine deck, currently ignoring vectorization
+    # of AviaryValues (use first index)
+    # TODO build test to verify this works as expected
+    engine_options = AviaryValues()
+    for entry in Aircraft.Engine.__dict__:
+        var = getattr(Aircraft.Engine, entry)
+        # check if this variable exist with useable metadata
+        try:
+            units = _MetaData[var]['units']
+            try:
+                # add value from aviary_options to engine_options
+                aviary_val = aviary_options.get_val(var, units)
+                if isinstance(aviary_val, np.ndarray):
+                    # "Convert" numpy types to standard Python types. Wrap first
+                    # index in numpy array before calling item() to safeguard against
+                    # non-standard types, such as objects
+                    aviary_val = aviary_val[0].item()
+                elif isinstance(aviary_val, (list, tuple)):
+                    aviary_val = aviary_val[0]
+                engine_options.set_val(var, aviary_val, units)
+            # if not, use default value from _MetaData
+            except KeyError:
+                # engine_options.set_val(var, _MetaData[var]['default_value'], units)
```

Comment:
I'm still not sure if we want this or not - right now it isn't needed because all our tests are well-defined, but we might want to default variables users don't input for engine decks. However, it's equally reasonable to not automatically add in variables the user didn't specify, since there's a chance that has unforseen consequences.

Right now the setup is if you need a var but didn't provide it things break, rather than using a default. Is that a good idea? I'm not sure

Reply:
In general I like having defaults, like having the scale factors default to 1, to reduce workload on users, especially if we want this to be accessible to new users/users of the legacy codes.

Maybe we can set default values for some (but not all) of the engine related variables, and use the set_val only if there is some default value.

Reply:
just noticed that the comments I left weren't marked as part of my latest review